### PR TITLE
Added framework and channel related API

### PR DESCRIPTION
**Pull request by arati-tekdi**
## Summary by CodeRabbit

- **New Features**
	- Enhanced API route configurations with improved role checks and flexibility in route creation.
	- New routes added for channel and framework APIs.

- **Bug Fixes**
	- Improved error response handling in the gateway service based on new conditions.

- **Refactor**
	- Code formatting adjustments for consistency and readability across multiple files.
	- Simplified middleware request processing and enhanced routing capabilities, including new service mappings.

### CODE REVIEW

File: src/middleware/gateway.service.ts [hunk 1]
```diff
@@ -4,13 +4,14 @@ import { MiddlewareLogger } from 'src/common/loggers/logger.service';
...
 export class GatewayService {
-  constructor(private readonly middlewareLogger: MiddlewareLogger) { }
+  constructor(private readonly middlewareLogger: MiddlewareLogger) {}
 
   async handleRequest(
     method: string,
     url: string,
     body: Object,
     oheaders: any,
+    changeResponse: boolean,
```
[the remark this answers is not in the document]
Comment (arati-tekdi) [reply src/middleware/gateway.service.ts:14]: Not needed

File: src/middleware/gateway.service.ts [hunk 2]
```diff
@@ -31,6 +32,22 @@ export class GatewayService {
...
+          ) {
+            error.response.data.responseCode = 'OK';
+            error.response.data.result = {
+              license: {
+                valid: false,
+                value: 'youtube',
+              },
+            };
+          }
+        }
```
[the remark this answers is not in the document]
Comment (arati-tekdi) [reply src/middleware/gateway.service.ts]: Done

File: src/middleware/gateway.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (arati-tekdi) [reply src/middleware/gateway.service.ts:14]: Not needed

File: src/middleware/gateway.service.ts [hunk 3]
```diff
@@ -31,6 +32,22 @@ export class GatewayService {
       return response.data;
     } catch (error) {
       if (error.response) {
+        if (changeResponse) {
+          if (
+            error.response.data.params.err ===
+              'ERR_YOUTUBE_LICENSE_VALIDATION' &&
+            url ===
+              'https://content-dev.prathamdigital.org/asset/v4/validate?field=license'
```
Comment (rushi-tekdi) [inline src/middleware/gateway.service.ts:40]: @arati-tekdi can you get this URL from env ?

Comment (arati-tekdi) [reply src/middleware/gateway.service.ts:40]: no. this is handling for one API. In future we will make that as generic. For now it is only for particular error, therefor i have added like this.

**Pull request merged**